translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+``admin.en.yml``        ``admin+intl-icu.en.yml``
+======================  ===============================
+
+All messages in this file will now be processed by the
+:phpclass:`MessageFormatter` during translation.
+
+.. _component-translation-placeholders:
+
+Message Placeholders
+--------------------
+
+The basic usage of the MessageFormat allows you to use placeholder (called
+*arguments* in ICU MessageFormat) in your messages:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        say_hello: 'Hello {name}!'
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="say_hello">
+                        <source>say_hello</source>
+                        <target>Hello {name}!</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'say_hello' => "Hello {name}!",
+        ];
+
+Everything within the curly braces (``{...}``) is processed by the formatter
+and replaced by it's placeholder::
+
+    // ...
+    echo $translator->trans('say_hello', ['name' => 'Fabien']); // Hello Fabien!
+    echo $translator->trans('say_hello', ['name' => 'Symfony']); // Hello Symfony!
+
+Selecting Different Messages Based on a Condition
+-------------------------------------------------
+
+The curly brace syntax allows to "modify" the output of the variable. One of
+these functions is the ``switch`` function. It acts like PHP's `switch statement`_
+and allows to use different strings based on the value of the variable. A
+typical usage of this is gender:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        invitation_title: >
+            {organizer_gender, select,
+                female {{organizer_name} has invited you for her party!}
+                male   {{organizer_name} has invited you for his party!}
+                other  {{organizer_name} have invited you for their party!}
+            }
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="invitation_title">
+                        <source>invitation_title</source>
+                        <target>{organizer_gender, select, female {{organizer_name} has invited you for her party!} male   {{organizer_name} has invited you for his party!} other {{organizer_name} have invited you for their party!}}</target>
```

Comment:
Maybe its just a rendering problem, but it looks like there are many whitespaces after `male` and not after `other`